cmd/kubeless/function/function.go

```diff
@@ -157,47 +157,49 @@ func getFunctionDescription(cli kubernetes.Interface, funcName, ns, handler, fro
 		function.Spec.Handler = handler
 	}
 
-	if fromURL != "" && file != "" {
-		err := errors.New("either --from-url or --from-file should be provided, not both")
-		return nil, err
-	}
-
-	if fromURL != "" {
-		resp, err := http.Get(fromURL)
-		if err != nil {
-			return nil, err
-		}
-		defer resp.Body.Close()
-
-		functionBytes, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return nil, err
-		}
-
-		checksum, err := getSha256(functionBytes)
-		if err != nil {
-			return nil, err
-		}
-		function.Spec.Checksum = checksum
+	if file != "" {
 
-		function.Spec.FunctionContentType = "url"
-		function.Spec.Function = fromURL
-	}
+		if strings.Index(file, "http://") == 0 || strings.Index(file, "https://") == 0 {
+			functionURL, err := url.Parse(file)
+			if err != nil {
+				return nil, err
+			}
+			resp, err := http.Get(functionURL.String())
+			if err != nil {
+				return nil, err
+			}
+			defer resp.Body.Close()
+
+			functionBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			function.Spec.FunctionContentType = "url"
+			// strip any query params to determine if the file is a zip
+			if path.Ext(strings.Split(functionURL.String(), "?")[0]) == ".zip" {
```

Comment:
I would move the check `if strings.Index(file, "http://")` and `if path.Ext(strings.Split(functionURL.String(), "?")[0]) == ".zip" ` to the function `getContentType` of the `function.go` file so the same function always returns the full contentType